modules/imgproc/src/histogram.cpp

```diff
@@ -2449,17 +2449,18 @@ void cv::calcBackProject( InputArrayOfArrays images, const std::vector<int>& cha
                           double scale )
 {
     CV_INSTRUMENT_REGION()
-
+    if (hist.dims() != 3)
```

Comment:
It's better to compare like that: `if(hist.dims() <= 2)` to avoid histograms of higher dimensions.